src/Dfe.PlanTech.Web/Controllers/QuestionsController.cs

```diff
@@ -47,6 +52,28 @@ public async Task<IActionResult> GetQuestionBySlug(string sectionSlug,
         return await router.ValidateRoute(sectionSlug, questionSlug, this, cancellationToken);
     }
 
+    [LogInvalidModelState]
+    [HttpGet("question/preview/{questionId}")]
+    public async Task<IActionResult> GetQuestionPreviewById(string questionId,
+                                                            [FromServices] ContentfulOptions contentfulOptions,
+                                                            CancellationToken cancellationToken = default)
+    {
+        if (!contentfulOptions.UsePreview)
+            return new RedirectResult("/self-assessment");
+
+        var question = await _getEntityFromContentfulQuery.GetEntityById<Question>(questionId, cancellationToken) ??
```

Comment:
Would be good to move all this logic out somewhere else IMO but YMMV

Reply:
now that its only the question i dont think its necessary, what do you reckon?